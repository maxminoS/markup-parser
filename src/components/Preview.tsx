import { parseMarkdown, parseOrg } from "utils/Parsers";

type Props = {
  parseTo: string,
  content: string,
}

export const Preview = (props: Props) => {
  return (
      <div style={{
            border: "1px solid black",
            display: "inline-block",
            width: "96%",
            height: "85vh",
            overflowY: "auto"
          }}
           dangerouslySetInnerHTML={{
             __html: props.parseTo === "Markdown" ? parseMarkdown(props.content) :
                     props.parseTo === "Org Mode" ? parseOrg(props.content) : props.content
           }}
      >
      </div>
  );
}
