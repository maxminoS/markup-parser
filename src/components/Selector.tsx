import React from 'react';
import { FormControl, InputLabel, Select, MenuItem } from "@material-ui/core";

type Props = {
  parseTo: string,
  setParseTo: (parseType: string) => void,
}

export const Selector = (props: Props) => {
  const handleSelectChange = (e: React.ChangeEvent<{ value: unknown }>) => props.setParseTo(e.target.value as string);

  return (
    <FormControl variant="outlined">
      <InputLabel>Markup</InputLabel>
      <Select
        value={props.parseTo}
        onChange={handleSelectChange}
        label="Markup"
      >
        <MenuItem value="HTML">HTML</MenuItem>
        <MenuItem value="Markdown">Markdown</MenuItem>
        <MenuItem value="Org Mode">Org Mode</MenuItem>
      </Select>
    </FormControl>
  );
}
