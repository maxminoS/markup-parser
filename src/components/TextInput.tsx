import React from 'react';

type Props = {
  content: string,
  setContent: (content: string) => void,
}

export const TextInput = (props: Props) => {
  const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => props.setContent(e.target.value);

  return(
    <textarea
          onChange={handleChange}
          style={{
              width: "98%",
              height: "85vh"
          }}
          value={props.content}
    />
  );
}
