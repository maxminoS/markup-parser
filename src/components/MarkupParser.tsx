import React, { useState } from "react";
import { AppBar, Toolbar, Grid, Box, Typography } from "@material-ui/core";

import { Preview } from "components/Preview";
import { TextInput } from "components/TextInput";
import { Selector } from "components/Selector";

export const MarkupParser = () => {
  const [content, setContent] = useState("");
  const [parseTo, setParseTo] = useState("HTML");

  return (
    <>
      <AppBar position="static" color="primary">
        <Toolbar style={{ minHeight: "75px" }}>
          <Box display="flex" flexGrow={1}>
            <Typography variant="h6" align="center">Markup Parser</Typography>
          </Box>
          <Selector parseTo={parseTo} setParseTo={setParseTo} />
        </Toolbar>
      </AppBar>
      <br />
      <Grid container
            justify="center"
            alignItems="flex-start"
            style={{ minHeight: "88vh" }}
            spacing={2}
      >
        <Grid item xs={5}>
          <TextInput content={content} setContent={setContent} />
        </Grid>
        <Grid item xs={5}>
          <Preview parseTo={parseTo} content={content} />
        </Grid>
      </Grid>
    </>
  );
}
