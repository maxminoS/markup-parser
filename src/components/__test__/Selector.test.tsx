import React from 'react';
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import { Selector } from "components/Selector";

describe("Selector", () => {
  it("has HTML option", () => {
    render(<Selector parseTo={"HTML"} setParseTo={() => {}} />);
    expect(screen.getByRole("button", { name: /html/i })).toBeInTheDocument();
  });

  it("has Markdown option", () => {
    render(<Selector parseTo={"Markdown"} setParseTo={() => {}} />);
    expect(screen.getByRole("button", { name: /markdown/i })).toBeInTheDocument();
  });

  it('has Org Mode option', () => {
    render(<Selector parseTo={"Org Mode"} setParseTo={() => {}} />);
    expect(screen.getByRole("button", { name: /org mode/i })).toBeInTheDocument();
  });
});

it('renders Selector component as expected', () => {
  const selector = renderer.create(<Selector parseTo={"HTML"} setParseTo={() => {}} />).toJSON();
  expect(selector).toMatchSnapshot();
});
