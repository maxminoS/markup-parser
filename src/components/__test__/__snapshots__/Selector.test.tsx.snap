// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Selector component as expected 1`] = `
<div
  className="MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
    data-shrink={true}
  >
    Markup
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      HTML
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="HTML"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-13 MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="PrivateNotchedOutline-legendLabelled-15 PrivateNotchedOutline-legendNotched-16"
      >
        <span>
          Markup
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
